irohad/network/impl/async_grpc_client.hpp

```diff
@@ -35,14 +34,18 @@ namespace iroha {
         void *got_tag;
         auto ok = false;
         while (cq_.Next(&got_tag, &ok)) {
-          auto call = static_cast<AsyncClientCall *>(got_tag);
+          std::unique_ptr<AsyncClientCall> call(
```

Comment:
I see call created bellow with `new` is removed here. This is reinterpret_cast, not static and it looks bad.

Isn't there a way to keey strict types?